docs/documentation/development/guidelines.md

```diff
@@ -26,7 +26,13 @@ To speed up the contribution process, we therefore advice to go through this che
 1. For dependency injection, OSGi Declarative Services should be used.
 1. OSGi Declarative Services should be declared using annotations. The IDE will take care of the service *.xml file creation. See the official OSGi documentation for an [example here](http://enroute.osgi.org/services/org.osgi.service.component.html). We always use `@Activate`, `@Deactivate` and `@Modified` if we define these methods, even if they exist in a super class, to make the code more readable.
 1. Packages that contain classes that are not meant to be used by other bundles should have "internal" in their package name.
-1. [Null annotations](https://wiki.eclipse.org/JDT_Core/Null_Analysis) are used from the Eclipse JDT project. Therefore every bundle should have an **optional** `Import-Package` dependency to `org.eclipse.jdt.annotation`.
+1. [Null annotations](https://wiki.eclipse.org/JDT_Core/Null_Analysis) are used from the Eclipse JDT project.
+Our intention with these annotations is to transfer a method's contract written in its JavaDoc into the code to be processed by tools.
+We are aware that these annotations can be used for **static** checks, but **not** at runtime.  
+Thus for publicly exposed methods that belong to our API and are (potentially) called by external callers, we cannot omit a `null` check although a method parameter is marked to be not `null` via an annotation.
+We will get a warning in the IDE for this check, but we decided to live with that.
+For private methods or methods in an internal package we agreed to respect the annotations and omit an additional `null` check.  
+To use the annotaions, every bundle should have an **optional** `Import-Package` dependency to `org.eclipse.jdt.annotation`.
```

Comment:
Spotted little typo: annotaions -> annotations

Reply:
@IOOOTAlan Thanks, I have fixed it in #5363